### PR TITLE
Fix the code style issues reported by elvis.

### PR DESCRIPTION
**Pull request by epretha**
This PR addresses 

Proposed changes include:
* Refactoring of ejabberd_auth*.erl modules to allow them to comply with elvis.config specifications.

### CODE REVIEW

File: apps/ejabberd/src/ejabberd_auth_internal.erl [hunk 1]
```diff
@@ -412,3 +392,23 @@ write_passwd(#passwd{} = Passwd) ->
...
+    case scram:enabled(LServer) and is_binary(Password) of
+        true ->
+            scram:password_to_scram(Password, scram:iterations(LServer));
+        false -> Password
+    end.
+
+-spec get_vh_registered_users_within_interval(list(), integer(), integer()) ->
+                                                    list().
+get_vh_registered_users_within_interval(Users, Limit, Offset) ->
+    case Users of
```
Comment (fenek) [inline apps/ejabberd/src/ejabberd_auth_internal.erl]: Please match it in function head.

File: apps/ejabberd/src/ejabberd_auth_ldap.erl [hunk 2]
```diff
@@ -150,16 +150,15 @@ authorize(Creds) ->
...
                      Password :: binary()) -> boolean().
 check_password(LUser, LServer, Password) ->
-    if Password == <<"">> -> false;
-       true ->
-           case catch check_password_ldap(LUser, LServer, Password)
-               of
-             {'EXIT', _} -> false;
-             Result -> Result
-           end
+    case Password of
```
Comment (fenek) [inline apps/ejabberd/src/ejabberd_auth_ldap.erl]: Please match it in function head.

File: apps/ejabberd/src/ejabberd_auth.erl [hunk 3]
```diff
@@ -564,22 +565,12 @@ entropy(IOList) ->
     case binary_to_list(iolist_to_binary(IOList)) of
         "" ->
             0.0;
-        S ->
+        EntropyList ->
```
[the remark this answers is not in the document]
Comment (epretha) [reply apps/ejabberd/src/ejabberd_auth.erl]: Sure Piotr. The code is now updated.

File: apps/ejabberd/src/ejabberd_auth.erl [hunk 4]
```diff
@@ -564,22 +565,12 @@ entropy(IOList) ->
...
-                               C >= $A, C =< $Z ->
-                                    [Digit, Printable, LowLetter, 26, Other];
-                               C >= 16#21, C =< 16#7e ->
-                                    [Digit, 33, LowLetter, HiLetter, Other];
-                               true ->
-                                    [Digit, Printable, LowLetter, HiLetter, 128]
-                            end
-                    end, [0, 0, 0, 0, 0], S),
-            length(S) * math:log(lists:sum(Set))/math:log(2)
+                    fun(EntropyContent, Acc) ->
```
[the remark this answers is not in the document]
Comment (epretha) [reply apps/ejabberd/src/ejabberd_auth.erl]: As it was an element from the ```EntropyList``` variable, I named it as ```EntropyContent```. As the list is now renamed to ```InputList```, the variable is now renamed to ```InputContent```.

File: apps/ejabberd/src/ejabberd_auth.erl [hunk 5]
```diff
@@ -600,13 +591,16 @@ auth_modules() ->
...
-    Methods = if
-                  Method == undefined -> [];
-                  is_list(Method) -> Method;
-                  is_atom(Method) -> [Method]
-              end,
+    Methods = get_auth_method_as_a_list(Method),
     [list_to_atom("ejabberd_auth_" ++ atom_to_list(M)) || M <- Methods].
 
+get_auth_method_as_a_list(AuthMethod) ->
+    case AuthMethod of
```
[the remark this answers is not in the document]
Comment (epretha) [reply apps/ejabberd/src/ejabberd_auth.erl]: Sure. This is now changed.

File: apps/ejabberd/src/ejabberd_auth.erl [hunk 6]
```diff
@@ -564,22 +565,12 @@ entropy(IOList) ->
...
-                                    [Digit, Printable, LowLetter, 26, Other];
-                               C >= 16#21, C =< 16#7e ->
-                                    [Digit, 33, LowLetter, HiLetter, Other];
-                               true ->
-                                    [Digit, Printable, LowLetter, HiLetter, 128]
-                            end
-                    end, [0, 0, 0, 0, 0], S),
-            length(S) * math:log(lists:sum(Set))/math:log(2)
+                    fun(EntropyContent, Acc) ->
+                            convert_to_ordered_content(EntropyContent, Acc)
```
[the remark this answers is not in the document]
Comment (epretha) [reply apps/ejabberd/src/ejabberd_auth.erl]: Sorry about it Piotr. The function is now renamed to ```get_type_information```.

File: apps/ejabberd/src/ejabberd_auth.erl [hunk 7]
```diff
@@ -631,13 +625,29 @@ authorize_with_check_password(Module, Creds) ->
...
-               Digest == undefined orelse DigestGen == undefined ->
+               _ when Digest == undefined orelse DigestGen == undefined ->
                    [LUser, LServer, Password]
            end,
     case erlang:apply(Module, check_password, Args) of
         true -> {ok, mongoose_credentials:set(Creds, auth_module, Module)};
         false -> {error, not_authorized}
     end.
+
+-spec convert_to_ordered_content(integer() | term(), list()) -> list().
```
[the remark this answers is not in the document]
Comment (epretha) [reply apps/ejabberd/src/ejabberd_auth.erl]: Sorry Piotr. The type ```term()``` is not needed. I have now removed it.

File: apps/ejabberd/src/ejabberd_auth.erl [hunk 8]
```diff
@@ -631,13 +625,29 @@ authorize_with_check_password(Module, Creds) ->
...
+convert_to_ordered_content(EntropyContent, [Digit, Printable, LowLetter,
+                                            HiLetter, Other]) ->
+    case EntropyContent of
```
[the remark this answers is not in the document]
Comment (epretha) [reply apps/ejabberd/src/ejabberd_auth.erl]: Refactored.

File: apps/ejabberd/src/ejabberd_auth_internal.erl (hunk 1 above)
Comment (epretha) [reply apps/ejabberd/src/ejabberd_auth_internal.erl]: Refactored.

File: apps/ejabberd/src/ejabberd_auth_ldap.erl (hunk 2 above)
Comment (epretha) [reply apps/ejabberd/src/ejabberd_auth_ldap.erl]: Refactored.

File: apps/ejabberd/src/ejabberd_auth.erl [hunk 9]
```diff
@@ -631,13 +625,29 @@ authorize_with_check_password(Module, Creds) ->
     Password  = mongoose_credentials:get(Creds, password),
     Digest    = mongoose_credentials:get(Creds, digest, undefined),
     DigestGen = mongoose_credentials:get(Creds, digest_gen, undefined),
-    Args = if
-               Digest /= undefined andalso DigestGen /= undefined ->
+    Args = case {Digest, DigestGen} of
+               _ when Digest /= undefined andalso DigestGen /= undefined ->
                    [LUser, LServer, Password, Digest, DigestGen];
-               Digest == undefined orelse DigestGen == undefined ->
+               _ when Digest == undefined orelse DigestGen == undefined ->
```
[the remark this answers is not in the document]
Comment (epretha) [reply apps/ejabberd/src/ejabberd_auth.erl]: True Piotr. The when guard is now removed for this clause.

**Pull request merged**